examples/plot_metricframe_beyond_binary_classification.py

```diff
@@ -0,0 +1,249 @@
+print(mf2.by_group)
+
+
+# %%
+# Non-scalar Inputs
+# =================
+#
+# :class:`~fairlearn.metrics.MetricFrame` does not require
+# its inputs to be scalars either. To demonstrate this, we
+# will use an image recognition example (kindly supplied by
+# Ferdane Bekmezci, Hamid Vaezi Joze and Samira Pouyanfar).
+#
+# Image recognition algorithms frequently construct a bounding
+# box around regions where they have found their target features.
+# For example, if an algorithm detects a face in an image, it
+# will place a bounding box around it. These bounding boxes
+# constitute `y_pred` for :class:`~fairlearn.metrics.MetricFrame`.
+# The `y_true` values then come from bounding boxes marked by
+# human labellers.
+#
+# Bounding boxes are often compared using the 'iou' metric.
+# This computes the intersection and the union of the two
+# bounding boxes, and returns the ratio of their areas.
+# If the bounding boxes are identical, then the metric will
+# be 1; if disjoint then it will be 0. A function to do this is:
+
+def bounding_box_iou(box_A_input, box_B_input):
+    # The inputs are array-likes in the form
+    # [x_0, y_0, delta_x,delta_y]
+    # where the deltas are positive
+
+    box_A = np.array(box_A_input)
+    box_B = np.array(box_B_input)
+
+    assert box_A[2] >= 0, "Bad delta_x for box_A"
+    assert box_A[3] >= 0, "Bad delta y for box_A"
+    assert box_B[2] >= 0, "Bad delta x for box_B"
+    assert box_B[3] >= 0, "Bad delta y for box_B"
+
+    # Convert deltas to co-ordinates
+    box_A[2:4] = box_A[0:2] + box_A[2:4]
+    box_B[2:4] = box_B[0:2] + box_B[2:4]
+
+    # Determine the (x, y)-coordinates of the intersection rectangle
+    x_A = max(box_A[0], box_B[0])
+    y_A = max(box_A[1], box_B[1])
+    x_B = min(box_A[2], box_B[2])
+    y_B = min(box_A[3], box_B[3])
+
+    if (x_B < x_A) or (y_B < y_A):
+        return 0
+
+    # Compute the area of intersection rectangle
+    interArea = (x_B - x_A) * (y_B - y_A)
+
+    # Compute the area of both the prediction and ground-truth
+    # rectangles
+    box_A_area = (box_A[2] - box_A[0]) * (box_A[3] - box_A[1])
+    box_B_area = (box_B[2] - box_B[0]) * (box_B[3] - box_B[1])
+
+    # Compute the intersection over union by taking the intersection
+    # area and dividing it by the sum of prediction + ground-truth
+    # areas - the intersection area
+    iou = interArea / float(box_A_area + box_B_area - interArea)
+
+    return iou
+
+# %%
+# This is a metric for two bounding boxes, but for :class:`~fairlearn.metrics.MetricFrame`
+# we need to compare two lists of bounding boxes. For the sake of
+# simplicity, we will return the mean value of 'iou' for the
+# two lists, but this is by no means the only choice:
+
+
+def mean_iou(true_boxes, predicted_boxes):
+    assert len(true_boxes) == len(predicted_boxes), "Array size mismatch"
+
+    all_iou = [
+        bounding_box_iou(true_boxes[i], predicted_boxes[i])
+        for i in range(len(true_boxes))
+    ]
```

Comment:
would you not prefer this? I think it should also be faster.

```suggestion
    all_iou = [
        bounding_box_iou(y_true, y_pred)
        for y_true, y_pred in zip(true_boxes, predicted_boxes)
    ]
```

Reply:
Changed